internal/client/client.go

```diff
@@ -101,6 +101,14 @@ func (c *CSAPI) SyncUntilTimelineHas(t *testing.T, roomID string, check func(gjs
 	c.syncUntil(t, "", "rooms.join."+gjsonEscape(roomID)+".timeline.events", check)
 }
 
+// SyncUntilKnockHas blocks and continually calls /sync until the `check` function returns true.
+// If the `check` function fails the test, the failing event will be automatically logged.
+// Will time out after CSAPI.SyncUntilTimeout.
+func (c *CSAPI) SyncUntilKnockHas(t *testing.T, roomID string, check func(gjson.Result) bool) {
```

Comment:
I don't think knocking is _that_ common that it should pollute the `CSAPI` client. Move this to the test file please.

Reply:
If you require `syncUntil` logic, make that public instead.